src/poetry/console/commands/source/add.py

```diff
@@ -86,7 +86,7 @@ def handle(self) -> Optional[int]:
                 )
                 return 1
 
-            if source.name == name:
+            if new_source and source.name == name:
```

Comment:
Without this check, then the below assignment `source = new_source` maybe have produced `source = None`